### DISCUSSION TITLE
How can we add an action to the global filament page

### DISCUSSION THREAD

**Discussion by StaffCollab** [filamentphp/filament#11522] [Help] [answered]
### Package

Panel builder

### Package Version

v3.2

### How can we help you?

I want to load an action instance on each and every filament page since I want to trigger it from within the global search via event. Is it possible easily?

**Comment by leandrocfe** [comment]
Could you explain more about it? What code did you try to achieve this?

**Reply by StaffCollab** [reply]
Yes. I want to use an action in global search that should open a model.

I'm trying to use global search actions, but they only support URL or event actions, not models. I want to fire an event and open a model from it, but since the global search is available globally, I need the action (the model that should be opened on the event) to be loaded on every page that has a global search.

~~~
public static function getGlobalSearchResultActions(Model $record): array
{
    return [
        Action::make('quickView')
            ->dispatch('quickView', [$record->id])
    ];
}
~~~
quickView event should now open the quickView action model, but of course, it needs to be globally available.

**Comment by ArtMin96** [comment, marked as answer]
[Render hooks](https://filamentphp.com/docs/3.x/support/render-hooks)

**Reply by StaffCollab** [reply]
Thanks, you're totally correct.
